test/e2e/helpers.go

```diff
@@ -0,0 +1,125 @@
+package e2e
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/url"
+	"testing"
+
+	"github.com/openshift-knative/serverless-operator/test"
+	v1 "github.com/openshift/api/route/v1"
+	ioprometheusclient "github.com/prometheus/client_model/go"
+	"github.com/prometheus/common/expfmt"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	pkgTest "knative.dev/pkg/test"
+)
+
+func extracMetrictData(body io.Reader, metricName string) (*float64, error) {
```

Comment:
Consider returning the dereferenced value here to avoid dereferencing in client code as they'd actually have to do a nil check.

Reply:
the contract is that if there is an error you dont check the value but ok.

Reply:
Also there is a typo to fix.